From 0fd0828c41ea74308cb20d301d1284c7089fd3eb Mon Sep 17 00:00:00 2001
From: Mika Laitio <lamikr@pilppa.org>
Date: Fri, 10 May 2024 17:06:25 -0700
Subject: [PATCH 2/5] aotriton triton rocm home path update

- change the location of rocm_home
  to triton subproject
  (submodule on original repositories)

Signed-off-by: Mika Laitio <lamikr@pilppa.org>
---
 third_party/triton/.github/workflows/amd-offline-tests.yml  | 6 +++---
 third_party/triton/CMakeLists.txt                           | 6 +++++-
 third_party/triton/lib/Target/HSACO/HSACOTranslation.cpp    | 2 +-
 third_party/triton/python/triton/common/build.py            | 2 +-
 .../triton/python/triton/third_party/hip/CMakeLists.txt     | 4 ++--
 .../third_party/hip/lib/Target/HSACO/HSACOTranslation.cpp   | 2 +-
 third_party/triton/scripts/amd/lld.sh                       | 2 +-
 third_party/triton/scripts/amd/setup_rocm_libs.sh           | 6 +++---
 8 files changed, 17 insertions(+), 13 deletions(-)

diff --git a/third_party/triton/.github/workflows/amd-offline-tests.yml b/third_party/triton/.github/workflows/amd-offline-tests.yml
index fd87f8f..2b7ccc3 100644
--- a/third_party/triton/.github/workflows/amd-offline-tests.yml
+++ b/third_party/triton/.github/workflows/amd-offline-tests.yml
@@ -36,10 +36,10 @@ jobs:
           apt-get install -y ./amdgpu-install_5.5.50500-1_all.deb
           amdgpu-install -y --usecase=rocm --no-dkms
           # remove unused packages to free some disk space
-          cp /opt/rocm/llvm/bin/ld.lld .
+          cp /opt/rocm_sdk_612/bin/ld.lld .
           apt purge -y rocblas rocm-llvm
-          mkdir -p /opt/rocm/llvm/bin
-          mv ld.lld /opt/rocm/llvm/bin/ld.lld
+          mkdir -p /opt/rocm_sdk_612/bin
+          mv ld.lld /opt/rocm_sdk_612/bin/ld.lld
           # install pytorch
           pip3 install torch==2.0.1 --index-url https://download.pytorch.org/whl/rocm5.4.2
           # uninstall system triton to eliminate conflicts with testing version
diff --git a/third_party/triton/CMakeLists.txt b/third_party/triton/CMakeLists.txt
index 55a5d8b..e880296 100644
--- a/third_party/triton/CMakeLists.txt
+++ b/third_party/triton/CMakeLists.txt
@@ -35,7 +35,7 @@ set(TRITON_CODEGEN_BACKENDS "" CACHE STRING "Enable different codegen backends")
 
 # Force TRITON_USE_ROCM for ROCm support
 set(TRITON_USE_ROCM ON)
-set(ROCM_DEFAULT_DIR "/opt/rocm")
+set(ROCM_DEFAULT_DIR "/opt/rocm_sdk_612")
 add_definitions( -DROCM_DEFAULT_DIR="${ROCM_DEFAULT_DIR}")
 
 # Ensure Python3 vars are set correctly
@@ -211,6 +211,10 @@ if(EXISTS "$ENV{ROCM_PATH}/lib/libhsa-runtime64.so" )
   set(ROCM_LIBRARIES
     "$ENV{ROCM_PATH}/lib/libhsa-runtime64.so"
   )
+elseif(EXISTS "$ENV{ROCM_PATH}/lib64/libhsa-runtime64.so" )
+  set(ROCM_LIBRARIES
+    "$ENV{ROCM_PATH}/lib64/libhsa-runtime64.so"
+  )
 elseif(EXISTS "${ROCM_DEFAULT_DIR}/lib/libhsa-runtime64.so" )
   set(ROCM_LIBRARIES
     "${ROCM_DEFAULT_DIR}/lib/libhsa-runtime64.so"
diff --git a/third_party/triton/lib/Target/HSACO/HSACOTranslation.cpp b/third_party/triton/lib/Target/HSACO/HSACOTranslation.cpp
index e6d36ad..1f5c31a 100644
--- a/third_party/triton/lib/Target/HSACO/HSACOTranslation.cpp
+++ b/third_party/triton/lib/Target/HSACO/HSACOTranslation.cpp
@@ -237,7 +237,7 @@ std::string generate_hsaco(llvm::Module *module, const std::string &triple,
   if (!std::filesystem::exists(lld_path)) {
     std::string rocm_path = ::triton::tools::getenv("ROCM_PATH");
     lld_path = (rocm_path.empty()) ? ROCM_DEFAULT_DIR : rocm_path;
-    lld_path += "/llvm/bin/ld.lld";
+    lld_path += "/bin/ld.lld";
   }
 
   int lld_result =
diff --git a/third_party/triton/python/triton/common/build.py b/third_party/triton/python/triton/common/build.py
index 8cc87ea..d5551b7 100644
--- a/third_party/triton/python/triton/common/build.py
+++ b/third_party/triton/python/triton/common/build.py
@@ -85,7 +85,7 @@ def rocm_path_dir():
     if (os.path.exists(default_path+"/include/hip/hip_runtime.h")):
         return default_path
     else:
-        return os.getenv("ROCM_PATH", default="/opt/rocm")
+        return os.getenv("ROCM_PATH", default="/opt/rocm_sdk_612")
 
 
 @contextlib.contextmanager
diff --git a/third_party/triton/python/triton/third_party/hip/CMakeLists.txt b/third_party/triton/python/triton/third_party/hip/CMakeLists.txt
index 1c5977f..92a080c 100644
--- a/third_party/triton/python/triton/third_party/hip/CMakeLists.txt
+++ b/third_party/triton/python/triton/third_party/hip/CMakeLists.txt
@@ -2,7 +2,7 @@ list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
 
 # FLAGS
 message(STATUS "HIP_BACKEND_MODE = ${HIP_BACKEND_MODE}")
-set(ROCM_DEFAULT_DIR "/opt/rocm")
+set(ROCM_DEFAULT_DIR "/opt/rocm_sdk_612")
 add_definitions( -DROCM_DEFAULT_DIR="${ROCM_DEFAULT_DIR}")
 set(ROCM_LIBRARIES
   ${CMAKE_CURRENT_SOURCE_DIR}/lib/hsa/libhsa-runtime64.so
@@ -106,4 +106,4 @@ if(TRITON_BUILD_PYTHON_MODULE)
         DEPENDS ${HIP_BACKEND_PY_STAMP}
         COMMENT "Checking and applying modifications to hip_backend.py"
     )
-endif()
\ No newline at end of file
+endif()
diff --git a/third_party/triton/python/triton/third_party/hip/lib/Target/HSACO/HSACOTranslation.cpp b/third_party/triton/python/triton/third_party/hip/lib/Target/HSACO/HSACOTranslation.cpp
index bbeb5a2..7e79da2 100644
--- a/third_party/triton/python/triton/third_party/hip/lib/Target/HSACO/HSACOTranslation.cpp
+++ b/third_party/triton/python/triton/third_party/hip/lib/Target/HSACO/HSACOTranslation.cpp
@@ -223,7 +223,7 @@ std::string generate_hsaco(llvm::Module *module, const std::string &triple,
     std::string rocm_path = ::triton::tools::getenv("ROCM_PATH");
     // auto ROCM_DEFAULT_DIR = "/opt/rocm";
     lld_path = (rocm_path.empty()) ? ROCM_DEFAULT_DIR : rocm_path;
-    lld_path += "/llvm/bin/ld.lld";
+    lld_path += "/bin/ld.lld";
   }
 
   int lld_result =
diff --git a/third_party/triton/scripts/amd/lld.sh b/third_party/triton/scripts/amd/lld.sh
index fd78d62..93e860b 100644
--- a/third_party/triton/scripts/amd/lld.sh
+++ b/third_party/triton/scripts/amd/lld.sh
@@ -1 +1 @@
-${ROCM_PATH}/llvm/bin/ld.lld -flavor gnu -shared _empty.o -o _empty.hsaco
+${ROCM_PATH}/bin/ld.lld -flavor gnu -shared _empty.o -o _empty.hsaco
diff --git a/third_party/triton/scripts/amd/setup_rocm_libs.sh b/third_party/triton/scripts/amd/setup_rocm_libs.sh
index 60eaa30..c9df795 100755
--- a/third_party/triton/scripts/amd/setup_rocm_libs.sh
+++ b/third_party/triton/scripts/amd/setup_rocm_libs.sh
@@ -4,7 +4,7 @@ set -ex
 
 # Set ROCM_HOME if not set
 if [[ -z "${ROCM_HOME}" ]]; then
-    export ROCM_HOME=/opt/rocm
+    export ROCM_HOME=/opt/rocm_sdk_612
 fi
 
 # Check TRITON_ROCM_DIR is set
@@ -69,6 +69,6 @@ done
 cp -r $ROCM_HOME/include $TRITON_ROCM_DIR/
 
 # Copy linker
-mkdir -p $TRITON_ROCM_DIR/llvm/bin
-cp $ROCM_HOME/llvm/bin/ld.lld $TRITON_ROCM_DIR/llvm/bin/
+mkdir -p $TRITON_ROCM_DIR/bin
+cp $ROCM_HOME/llvm/bin/ld.lld $TRITON_ROCM_DIR/bin/
 
-- 
2.41.1

